python/graphstorm/gconstruct/utils.py

```diff
@@ -530,6 +541,117 @@ def to_tensor(self):
         """
         return th.tensor(self.to_numpy())
 
+class ExtFeatureWrapper(ExtMemArrayWrapper):
+    """ The wrapper to memory-mapped Numpy array when combining feature
+
+    Parameters
+    ----------
+    arr_path : str
+        The path to different feature files directory
+    shape : tuple
+        The shape of the array.
+    dtype : numpy dtype
+        The data type.
+    merged_file: str
+        The merged file name
+    """
+    def __init__(self, arr_path, shape, dtype,
+                 merged_file="merged_feature.npy"):
+        self._arr_path = arr_path
+        self._shape = (shape[0], 0)
+        self._orig_dtype = self._dtype = dtype
+        self._arr = None
+        self.merged_file = merged_file
+        self.wrapper = []
+
+    @property
+    def dtype(self):
+        """ The data type
+        """
+        return self._dtype
+
+    @property
+    def shape(self):
+        """ The shape of the array.
+        """
+        return self._shape
+
+    def astype(self, dtype):
+        """ Return an array with converted data type.
+        """
+        arr = copy.copy(self)
+        arr._dtype = dtype
+        return arr
+
+    def __len__(self):
+        return self._shape[0]
+
+    def __getitem__(self, idx):
+        if self._shape[1] == 0:
+            raise RuntimeError("The ExtFeatureWrapper needs to be merge first")
+        if self._arr is None:
+            self._arr = np.memmap(os.path.join(self._arr_path, self.merged_file),
+                                  self._orig_dtype, mode="r",
+                                  shape=self._shape)
+        return self._arr[idx].astype(self.dtype)
+
+    def __del__(self):
+        self.cleanup()
+
+    def cleanup(self):
+        """ Clean up the array.
+        """
+        self._arr.flush()
+        self._arr = None
+        shutil.rmtree(self._arr_path)
+
+    def to_numpy(self):
+        """ Convert the data to Numpy array.
+        """
+        if self._arr is None:
+            arr = np.memmap(os.path.join(self._arr_path, self.merged_file),
+                            self._orig_dtype, mode="r", shape=self._shape)
+            if self._dtype != self._orig_dtype:
+                arr = arr.astype(self._dtype)
+            return arr
+        else:
+            return self._arr.astype(self._dtype)
+
+    def to_tensor(self):
+        """ Return Pytorch tensor.
+        """
+        return th.tensor(self.to_numpy())
+
+    def append(self, wrap, path=None):
```

Comment:
The path is not necessary, we can generate a temporal path.
BTW, please add doc strings.

Reply:
Let me move the outside path inside so we will not need temporal path here.